esm.mjs

```diff
@@ -0,0 +1,194 @@
+/**
+ *  base64.mjs
+ *
+ *  Licensed under the BSD 3-Clause License.
+ *    http://opensource.org/licenses/BSD-3-Clause
+ *
+ *  References:
+ *    http://en.wikipedia.org/wiki/Base64
+ * 
+ * @author Dan Kogai (https://github.com/dankogai)
+ */      
+const version = '3.1.3';
+const _b64chars
+      = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/';
+const _b64tab = ((bin) => {
+    let tab = {}, i = 0;
+    for (const c of bin) tab[c] = i++;
+    return tab;
+})(_b64chars);
+const _fromCharCode = String.fromCharCode;
+const _mkUriSafe =  (src) => String(src)
+      .replace(/[+\/]/g, (m0) => m0 == '+' ? '-' : '_')
+      .replace(/=/g, '');
+/**
+ * converts a Uint8Array to a Base64 string
+ * @param {Uint8Array} src
+ * @param {Boolean} urisafe URL-and-filename-safe a la RFC4648
+ * @returns {String} Base64 string
+ */
+const fromUint8Array = (src, urisafe) => {
+    let b64 = '';
+    for (let i = 0, l = src.length; i < l; i += 3) {
+        const a0 = src[i], a1 = src[i+1], a2 = src[i+2];
+        const ord = a0 << 16 | a1 << 8 | a2;
+        b64 +=    _b64chars.charAt( ord >>> 18)
+            +     _b64chars.charAt((ord >>> 12) & 63)
+            + ( typeof a1 != 'undefined'
+                ? _b64chars.charAt((ord >>>  6) & 63) : '=')
+            + ( typeof a2 != 'undefined'
+                ? _b64chars.charAt( ord         & 63) : '=');
+    }
+    return urisafe ? _mkUriSafe(b64) : b64;
+};
+/**
+ * 100% compatible with `window.btoa` of web browsers
+ * @param {String} src binary string
+ * @returns {String} Base64-encoded string
+ */
+const _btoa = typeof btoa === 'function'
+      ? btoa
+      : (src) => {
+          if (src.match(/[^\x00-\xFF]/)) throw new RangeError(
+              'The string contains invalid characters.'
+          );
+          return fromUint8Array(
+              Uint8Array.from(src,c => c.charCodeAt(0))
+          );
+      };
+/**
+ * @deprecated since 3.0.0
```

Comment:
Wondering why this function is marked `deprecated` since `encode` function uses it actually.

Reply:
Yes it is used internally.  It should be not used externally but I was exposed for for some reason.